### PR TITLE
Task/tut reorg aws

### PR DESCRIPTION
**Pull request by mdavis36**
# Summary

- This PR reworks the tutorial section of the RTD RAJA User Guide and exercise/example files for the upcoming AWS tutorial.

The RTD generated documentation for this PR is here: https://raja.readthedocs.io/en/task-tut-reorg-aws/sphinx/user_guide/tutorial.html

### CODE REVIEW

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (rchen20) [inline docs/sphinx/user_guide/feature/loop_basic.rst:270]:
```suggestion
Hierarchical loops (RAJA::launch)
```

File: exercises/kernelintro-nested-loop-reorder_solution.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder_solution.cpp:145]: For consistency with other print statements.

```suggestion
  std::cout << "\n Running C-style nested loop order: I-outer, K-middle, J-inner" 
```

File: exercises/kernelintro-nested-loop-reorder_solution.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder_solution.cpp:108]:
```suggestion
  std::cout << "\n Running C-style nested loop order: J-outer, I-middle, K-inner" 
```

File: exercises/kernelintro-nested-loop-reorder_solution.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder_solution.cpp:70]:
```suggestion
  std::cout << "\n Running C-style nested loop order: K-outer, J-middle, I-inner" 
```

File: exercises/kernelintro-nested-loop-reorder_solution.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder_solution.cpp:18]:
```suggestion
 *  In this exercise, we introduce the basic RAJA::kernel mechanic for executing
 *  nested loop kernels, including using execution policies to permute the 
 *  order of loops in a loop nest. The exercise performs no actual 
```

File: exercises/kernelintro-nested-loop-reorder.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder.cpp:135]:
```suggestion
  std::cout << "\n Running C-style nested loop order: I-outer, K-middle, J-inner" 
```

File: exercises/kernelintro-nested-loop-reorder.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder.cpp:108]:
```suggestion
  std::cout << "\n Running C-style nested loop order: J-outer, I-middle, K-inner" 
```

File: exercises/kernelintro-nested-loop-reorder.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder.cpp:70]:
```suggestion
  std::cout << "\n Running C-style nested loop order: K-outer, J-middle, I-inner" 
```

File: exercises/kernelintro-nested-loop-reorder.cpp
Comment (rchen20) [inline exercises/kernelintro-nested-loop-reorder.cpp:18]:
```suggestion
 *  In this exercise, we introduce the basic RAJA::kernel mechanic for executing
 *  nested loop kernels, including using execution policies to permute the 
 *  order of loops in a loop nest. The exercise performs no actual 
```

File: examples/raja-launch.cpp
Comment (rchen20) [inline examples/raja-launch.cpp:20]:
```suggestion
 * Launch introduces hierarchical parallelism through the concept of
```

File: docs/sphinx/user_guide/tutorial/vertexsum_coloring.rst
Comment (rchen20) [inline docs/sphinx/user_guide/tutorial/vertexsum_coloring.rst:69]:
```suggestion
conditions where multiple threads may attempt to sum to a shared element 
```

**Comment by rhornung67** [comment]
Woo-hoo! all checks passing!!!

**Comment by rhornung67** [comment]
@mdavis36 looks like you fixed the azure CI. Thanks for fixing it quickly. Not sure why the Gitlab CI is reported as failing, while jobs are still pending.

**Comment by mdavis36** [comment]
> @mdavis36 looks like you fixed the azure CI. Thanks for fixing it quickly. Not sure why the Gitlab CI is reported as failing, while jobs are still pending.

I restarted the job, seemed like a random failure.

**Comment by rhornung67** [comment]
@mdavis36 @rchen20 @artv3 I think this PR is good to go. I've worked the remaining tutorial sections that we didn't get to before the AWS tutorial and did a final proofread pass (ran files through spellchecker, fixed broken links, etc.). Please look over and approve or note issues that need to be fixed. I am reworking the feature sections on another branch. Those should be ready to review soon after this one is merged.

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (artv3) [inline docs/sphinx/user_guide/feature/loop_basic.rst:328]: I noticed that we mix RAJA::expt::launch and RAJA::launch. But I think @ajkunen made a change so that we can use experimental features without using the expt:: name space?

**Review by artv3 — APPROVED** [review]

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/loop_basic.rst:328]: For the release, let's move launch stuff out of the 'expt' namespace and fix up the docs.

**Comment by mdavis36** [comment]
I can't approve the PR, but it looks good to me!

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**
